mypy/checker.py

```diff
@@ -587,10 +588,11 @@ def is_implicit_any(t: Type) -> bool:
 
                     # Refuse covariant parameter type variables
                     if isinstance(arg_type, TypeVarType):
-                        if arg_type.variance == COVARIANT:
-                            self.fail(messages.FUNCTION_PARAMETER_CANNOT_BE_COVARIANT,
-                                      arg_type)
-
+                        if i > 0:
+                            if arg_type.variance == COVARIANT:
```

Comment:
Add comment about needing to the type variable check recursively for the argument types. For example, `Iteratble[T]` is not a valid argument type if `T` is covariant.